### PR TITLE
typo and functional fixes

### PR DESCRIPTION
**Pull request by Larsundso**

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -271,17 +267,14 @@ export function createProxyCache<
...
       const user = await bot.helpers.getUser(userId);
 
       await bot.cache.users.set(user)
 
       return bot.cache.users.get(userId);
     },
-    fetchMessage: async (messageId: bigint, guildId: bigint, channelId: bigint) => {
-      if (!options.fetchIfMissing?.messages) return;
-      if (unavailablesGuilds.has(guildId)) return;
+    fetchMessage: async (messageId: bigint, channelId: bigint, guildId?: bigint | undefined) => {
```
Comment (AwesomeStickz) [inline index.ts]: useless `| undefined`

File: index.ts [hunk 2]
```diff
@@ -735,20 +738,23 @@ export function createProxyCache<
...
+        if (stored) return stored;
       }
 
-      if (stored) return stored;
-
-      if (options.fetchIfMissing?.messages && channelId && guildId) {
-        return fetchers.fetchMessage(messageID, BigInt(guildId), BigInt(channelId));
+      const channelID = channelId ?? bot.cache.messages.channelIDs.get(BigInt(id));
+      if (options.fetchIfMissing?.messages && channelID) {
+        const guildID = guildId ?? channelID ? bot.cache.channels.guildIDs.get(BigInt(channelID)) : undefined;
```
Comment (AwesomeStickz) [inline index.ts:788]: `guildId ?? channelID`?

**Comment by deepsarda** [comment]
@Larsundso please note #17.
Which fixes #16 and #14

File: index.ts (hunk 2 above)
Comment (Larsundso) [reply index.ts:788]: > `guildId ?? channelID`?

check the whole line
` const guildID = guildId ?? (channelID ? bot.cache.channels.guildIDs.get(BigInt(channelID)) : undefined);`

Comment (AwesomeStickz) [reply index.ts:788]: Somehow didn't see that

**Review by Skillz4Killz — APPROVED** [review]

**Comment by Larsundso** [comment]
something isnt right wth channel fetching, fixing before merge

**Comment by Larsundso** [comment]
PR can now be merged

**Pull request merged**